core/src/main/java/org/apache/brooklyn/core/mgmt/classloading/JavaBrooklynClassLoadingContext.java

```diff
@@ -133,17 +129,11 @@ public boolean equals(Object obj) {
 
     @Override
     public URL getResource(String name) {
-        return getClassLoader().getResource(name);
+        return new ClassLoaderUtils(loader).getResource(name);
     }
 
     @Override
     public Iterable<URL> getResources(String name) {
-        Enumeration<URL> resources;
-        try {
-            resources = getClassLoader().getResources(name);
-        } catch (IOException e) {
-            throw Exceptions.propagate(e);
-        }
-        return Collections.list(resources);
+        return new ClassLoaderUtils(loader).getResources(name);
```

Comment:
ClassLoader.getResources() should return an empty Iterable. At first blush, it looks like this might throw an exception, or perhaps return null?